src/cppmangle.c

```diff
@@ -164,7 +164,8 @@ class CppMangleVisitor : public Visitor
                         else
                         {
                             dinteger_t val = e->toInteger();
-                            if (val < 0)
+                            // TODO: try and do this without knowing what dinteger_t type is
+                            if ((long)val < 0)
```

Comment:
> Yeah you're right. It really should be sinteger_t though.

Yep, I will do. Updates coming soon...

Note: I also found a test `runnable/cppmangle1.d` which seems to be more appropriate place for mangling. I have added test to there and verified old compiler doesn't link and new compiler does.